spec/requests/v1/qc_reception_spec.rb

```diff
@@ -0,0 +1,266 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe '/qc_receptions' do
+  describe '#post' do
+    before do
+      # QcAssayType for Extraction - to be ignored
+      create(:qc_assay_type, key: 'qubit_concentration_ngul', label: 'Qubit DNA Quant (ng/ul) [ESP1]', used_by: 0, units: 'ng/ul')
+      # Relevant QcAssayType for ToL
+      create(:qc_assay_type, key: 'sheared_femto_fragment_size', label: 'Sheared Femto Fragment Size (bp)', used_by: 1, units: 'bp')
+    end
+
+    # Load config file
+    let(:config) { YAML.load_file(Rails.root.join('config/locales//en.yml'), aliases: true) }
+    let(:error_config) { config['en']['activemodel']['errors']['models']['qc_receptions_factory'] }
+    let(:qc_results_list) do
+      [
+        {
+          final_nano_drop: '200',
+          final_nano_drop_230: '230',
+          final_nano_drop_280: '280',
+          post_spri_concentration: '10',
+          post_spri_volume: '20',
+          sheared_femto_fragment_size: '5',
+          shearing_qc_comments: 'Comments',
+          labware_barcode: 'FD20706500',
+          sample_external_id: 'supplier_sample_name_DDD'
+        }
+      ]
+    end
+
+    let(:source) { 'tol-lab-share.tol' }
+
+    context 'with valid parameters' do
+      let(:body) do
+        {
+          data: {
+            type: 'qc_receptions',
+            attributes: {
+              source:,
+              qc_results_list:
+            }
+          }
+        }.to_json
+      end
+
+      it 'returns a created status' do
+        post v1_qc_receptions_url, params: body, headers: json_api_headers
+        expect(response).to have_http_status(:created)
+      end
+
+      it 'creates a new QcReception' do
+        expect do
+          post v1_qc_receptions_url, params: body, headers: json_api_headers
+        end.to change(QcReception, :count).by(1)
+      end
+
+      it 'creates a new QcReception with the given source' do
+        post v1_qc_receptions_url, params: body, headers: json_api_headers
+        expect(QcReception.last.source).to eq source
+      end
+
+      it 'creates the relevant QcResult' do
+        expect do
+          post v1_qc_receptions_url, params: body, headers: json_api_headers
+        end.to change(QcResult, :count).by(1)
```

Comment:
Maybe test where there is more than one QcResult being created

Reply:
Done